op-node/node/node.go

```diff
@@ -439,9 +437,12 @@ func (n *OpNode) initL2(ctx context.Context, cfg *Config) error {
 	}
 
 	if cfg.Rollup.ChainOpConfig == nil {
-		chainCfg, err := loadOrFetchChainConfig(ctx, cfg.Rollup.L2ChainID, rpcClient)
+		if !cfg.Rollup.L2ChainID.IsUint64() {
+			return fmt.Errorf("cfg.Rollup.ChainOpConfig is nil and cfg.Rollup.L2ChainID is not set to a valid chain ID, preventing lookup in superchain registry")
+		}
+		chainCfg, err := superutil.LoadOPStackChainConfigFromChainID(cfg.Rollup.L2ChainID.Uint64())
 		if err != nil {
-			return fmt.Errorf("failed to load or fetch chain config for id %v: %w", cfg.Rollup.L2ChainID, err)
+			return fmt.Errorf(`cfg.Rollup.ChainOpConfig is nil and could not load from the superchain registry. Please see https://github.com/ethereum-optimism/optimism/releases/tag/op-node/v1.11.0: %w`, err)
 		}
 		cfg.Rollup.ChainOpConfig = chainCfg.Optimism
 	}
```

Comment:
Let's go one step further and just return the error straight away `if cfg.Rollup.ChainOpConfig == nil`. The superchain loader of the rollup config should already populate that field.

Reply:
[8fd8d22](https://github.com/ethereum-optimism/optimism/pull/15041/commits/8fd8d2282dc8a5f6639c9d62eb89a7e18389bf21)